modules/examples/sources/msg-generator/README.md

```diff
@@ -0,0 +1,24 @@
+# Message generator source
+
+This example source plugin generates a message every `freq()` seconds based on its `template()` option.
+
+Please note that `template()` is a source-side template, only message-independent macros can be used.
+
+### Usage
+
+```
+source s_generator {
+  msg-generator(
+    template("-- ${ISODATE} ${LOGHOST} ${SEQNUM} Generated message. --")
+    freq(0.5)
+  );
+};
```

Comment:
this could actually become a useful source of its own. especially if it would be able to execute an external command...